### ISSUE TITLE
Auto Profile seems to need all three match columns to be used

### ISSUE THREAD

**Issue by darkoppressor** [Ryochan7/antimicro#227]
I am using one device, and several different profiles. I have a simple profile attached to the default auto profile, and then each other profile is attached to a window using the various matching criteria (Class, Title, and Program).

On each auto profile, I am only using one of the match criteria. Among my various profiles, the Class and Title match criteria are both used. However, I make no use of the Program match criterion.

With this setup, the default auto profile was always staying active. I found that I could change any one of my auto profiles to use the unused Program match criterion, and this would fix the problem.

So in my case, it seems that all three match criteria must be used in at least one auto profile, or antimicro is never finding any auto profile matches, thus defaulting to the default auto profile.

**Comment by Ryochan7** [comment]
I have found the root of the problem. The big issue is that the application file path is only obtained when any auto profile entry has an application path defined. Without an application path being defined, AutoProfileWatcher will just used a blank string as the application location and that would cause an extra property match that should not happen. Since the number of matched properties differs from the number of properties defined, matching will fail and the default auto profile entry will be used instead.

I have made a small change that should ensure that properties will only be matched when defined. The change has been tested and it does fix this problem. In the mean time, you will have to have at least one auto profile entry that uses the application path to match criteria. Thank you for bringing up this problem.
